src/Instantiator.php

```diff
@@ -79,20 +92,32 @@ public function withoutConstructor(): self
 
     /**
      * Ignore attributes that can't be set to object.
+     *
+     * @param string ...$attributes The attributes you'd like the instantiator to ignore (if empty, ignore any extra)
      */
-    public function allowExtraAttributes(): self
+    public function allowExtraAttributes(string ...$attributes): self
     {
-        $this->allowExtraAttributes = true;
+        if (empty($attributes)) {
+            $this->allowExtraAttributes = true;
+        }
+
+        $this->extraAttributes = $attributes;
 
         return $this;
     }
 
     /**
      * Always force properties, never use setters (still uses constructor unless disabled).
+     *
+     * @param string ...$properties The properties you'd like the instantiator to "force set" (if empty, force set all)
      */
-    public function alwaysForceProperties(): self
+    public function alwaysForceProperties(string ...$properties): self
```

Comment:
Variadic arguments are difficult to manage in a BC way (it's no longer possible to add any optional argument after `$properties` in the future). I would recommend to go for a simple `array $properties = []` argument.

Reply:
Good call.